src/cr/cube/measures/pairwise_pvalues.py

```diff
@@ -0,0 +1,93 @@
+# encoding: utf-8
+
+"""P-values of pairwise comparison or columns of a contingency table."""
+
+from __future__ import division
+
+import numpy as np
+
+from cr.cube.distributions.wishart import wishartCDF
+from cr.cube.util import lazyproperty
+
+try:
+    xrange
+except NameError:  # pragma: no cover
+    # pylint: disable=invalid-name
+    xrange = range
+
+
+# pylint: disable=too-few-public-methods
+class PairwisePvalues:
+    """Value object providing matrix of pairwise-comparison P-values"""
+
+    def __init__(self, slice_, axis=0, weighted=True):
+        self._slice = slice_
+        self._axis = axis
+        self._weighted = weighted
+
+    @lazyproperty
+    def values(self):
+        """Square ndarray of pairwise Chi-square along axis."""
+        return 1.0 - wishartCDF(self._pairwise_chisq, self._n_min, self._n_max)
+
+    @lazyproperty
+    def _categorical_pairwise_chisq(self):
+        """ndarray containing pairwise comparisons along axis
+
+        Returns a square, symmetric matrix of test statistics for the null
+        hypothesis that each vector along *axis* is equal to each other.
+        """
+        chisq = np.zeros([self._numel, self._numel])
+        for i in xrange(1, self._numel):
+            for j in xrange(0, self._numel - 1):
+                chisq[i, j] = chisq[j, i] = np.sum(
+                    np.square(self._proportions[:, i] - self._proportions[:, j])
+                    / self._wts
+                ) / (1 / self._margin[i] + 1 / self._margin[j])
+
+        return chisq
+
+    @lazyproperty
+    def _margin(self):
+        """ndarray of floats, representing slice's margin for the correct axis."""
+        return self._slice.margin(axis=self._axis)
+
+    @lazyproperty
+    def _off_margin(self):
+        """ndarray of floats, representing slice's margin for the opposite axis."""
+        return self._slice.margin(axis=(1 - self._axis))
+
+    @lazyproperty
+    def _pairwise_chisq(self):
+        """ndarray of pairwise Chi-squared statistics along axis.
+
+        Zscore is a measure of statistical significance of observed vs.
+        expected counts. It's only applicable to a 2D contingency tables.
+        """
+        return self._categorical_pairwise_chisq
+
+    @lazyproperty
+    def _proportions(self):
+        return self._slice.proportions(axis=self._axis)
+
+    @lazyproperty
+    def _n_max(self):
+        """zero based int representing the bigger of the two cube's dimension."""
+        return max(self._slice.get_shape()) - 1
+
+    @lazyproperty
+    def _n_min(self):
+        """zero based int representing the smaller of the two cube's dimension."""
+        return min(self._slice.get_shape()) - 1
+
+    @lazyproperty
+    def _numel(self):
+        """int number of elements of the dimension opposite to axis."""
+        return self._slice.get_shape()[1 - self._axis]
+
+    @lazyproperty
+    def _wts(self):
+        # TODO: @mike - come up with a better docstring and property name
```

Comment:
@malecki This property needs a better name and docstring (i.e. this is just a placeholder that I came up with based on the original name of the variable from within the function).